packages/ubuntu_desktop_installer/test/allocate_disk_space_page_test.dart

```diff
@@ -263,6 +263,41 @@ void main() {
     verify(model.reformatDisk(disk)).called(1);
   });
 
+  testWidgets('confirm new partition table', (tester) async {
+    const disk = Disk(ptable: 'gpt', path: '/dev/sda');
+    final model = buildModel(
+      disks: [disk],
+      selectedDisk: disk,
+      canReformatDisk: true,
+    );
+    await tester.pumpWidget(tester.buildApp((_) => buildPage(model)));
+
+    final resetButton = find.ancestor(
+      of: find.text(tester.lang.newPartitionTable),
+      matching: find.byType(OutlinedButton),
+    );
+    expect(resetButton, findsOneWidget);
+    expect(tester.widget<OutlinedButton>(resetButton).enabled, isTrue);
+
+    await tester.tap(resetButton);
+    await tester.pumpAndSettle();
+
+    expect(find.byType(AlertDialog), findsOneWidget);
+    verifyNever(model.reformatDisk(disk));
+
+    final continueButton = find.widgetWithText(
+      OutlinedButton,
+      tester.ulang.continueAction,
+    );
+    expect(continueButton, findsNWidgets(2));
+
+    await tester.tap(continueButton.last);
```

Comment:
Are we guaranteed that this will always be the right continue button?

Reply:
Good point. I made it to explicitly look for an ancestor of AlertDialog.